org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/ConfigEntry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Inherited
+@Target({
+	ANNOTATION_TYPE
+})
+@Retention(RUNTIME)
+@Documented
+public @interface ConfigEntry {
+
+	String key();
+
+	String value() default "";
+
+}
```

Comment:
I'm not sure if this annotation adds much value. Given that it only accepts strings anyway, it would probably be easier if the `@Configuration.properties()` was simply a `String[]`, with each string of the form "name=value". This is how properties are specified using DS injection, for example.

Reply:
Using `"name=value"` is kind of a hack. In DS we now prefer ComponentPropertyType since the annotation is user defined and type safe. But that may be a lot for testing to support that. The one thing missing in ConfigEntry is to support the property type since CM supports more that String values.

Reply:
Using annotations for type safety would make sense, but as you noted at the moment `@ConfigEntry` only supports strings. If we're going to use the annotation so we can get type safety in the future, a bit of a simple roadmap might be a good idea. 

This is actually a similar problem space to the sources that Jupiter's parameterised tests use. Perhaps we could look there for inspiration.

Reply:
added a type to the annotation